plugins/gateway/tyk/v3.2.2/plugin.go

```diff
@@ -62,72 +78,104 @@ func ResponseSendTelemetry(rw http.ResponseWriter, res *http.Response, req *http
 		return
 	}
 
-	apiClient := utils.NewAPIClient("apiclarity.apiclarity:9000")
+	apiClient := newAPIClient(host)
 	params := operations.NewPostTelemetryParams().WithBody(telemetry)
 
 	_, err = apiClient.Operations.PostTelemetry(params)
 	if err != nil {
 		logger.Errorf("Failed to post telemetry: %v", err)
 		return
 	}
-	B, _ := json.Marshal(telemetry)
-	logger.Infof("Telemetry has been sent: %s", B)
+	logger.Infof("Telemetry has been sent")
 }
 
 func createTelemetry(res *http.Response, req *http.Request) (*models.Telemetry, error) {
+	truncatedBodyReq := false
+	truncatedBodyRes := false
+
 	apiDefinition := ctx.GetDefinition(req)
 	if apiDefinition == nil {
-		return nil, fmt.Errorf("failed to get api definition in response")
+		return nil, fmt.Errorf("failed to get api definition")
 	}
 
 	host, port := getHostAndPortFromTargetURL(apiDefinition.Proxy.TargetURL)
-	var destinationNamespace string
-	sp := strings.Split(host, ".")
-	if len(sp) == 2 {
-		destinationNamespace = sp[0]
-	}
+	destinationNamespace := getDestinationNamespaceFromHost(host)
 
 	reqBody, err := ioutil.ReadAll(req.Body)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read request body: %v", err)
 	}
+	if len(reqBody) > MaxBodySize {
```

Comment:
better to do it in a function (that returns the new body and whether was truncation) and reuse in req&res and also to add a unit test